src/Http/ServerRequest.php

```diff
@@ -238,7 +230,6 @@ public function __construct(array $config = [])
             'base' => '',
             'webroot' => '',
             'input' => null,
-            'mergeFilesAsObjects' => true,
```

Comment:
I can retain just the merging of files data into `ServerRequest::$data` as either `UploadedFileInterface` instances or array in `ServerRequest` if that's preferable.